docs/PFC-test-plan.md

```diff
@@ -0,0 +1,200 @@
+unreliability. Transmitters can send packets faster than receivers can
+accept packets. As a result the receivers\' buffer keeps growing and
+beyond a threshold it starts silently dropping incoming packets. Flow
+control functions at Layer 2 offers a solution to this problem. Flow
+control enables feedback from a receiver to its sender to communicate
+buffer availability. IEEE 802.3x PAUSE control frames are introduced to
+achieve Layer 2 flow control. A receiver can send a PAUSE request to a
+sender when it senses potential buffer overflow. Upon receiving it, the
+sender stops transmission of any new packets on the wire until the
+receiver is ready to accept them again.
+
+IEEE 802.3x PAUSE works as designed but imposes certain limitations.
+Once a link is paused, the sender cannot send any traffic on that. As a
+result, an Ethernet Segment that carries multiple application flows with
+different QoS requirement, gets entirely blocked irrespective of their
+QoS. Thus enabling PAUSE for one application can affect the performance
+of the other network applications sharing the same Ethernet segment.
+IEEE 802.1Qbb PFC (Priority Flow Control) extends the basic PAUSE
+mechanism for multiple CoS, thus enabling coexistence of flows which
+needs flow control with other flows which performs better without it.
+Upto 8 CoSs can be used. PFC uses the same 64Byte MAC Control frame
+format that PAUSE frames do. It has a two byte Class-Enable vector to
+mention the CoS value for which the PAUSE should apply. As PFC acts
+independently on eight different CoSs, the frame describes the PAUSE
+duration for each CoS. The pause duration for each priority is a 2-byte
+value that expresses time as a number of quanta, where each quanta
+represents the time needed to transmit 512 bits at the current network
+speed. For user data traffic, the CoS maps to either CoS values defined
+in 802.1Q VLAN tag or the IP DSCP values.
+
+SONiC has two lossless priorities: 3 and 4, by default. It is to be
+noted that only the lossless priorities can react to or generate PFC
+frames. In other words, PFC frames should not have any impact on traffic
+on lossy priorities. Packets with Differentiated Services Code Point
+(DSCP) 3 and 4 are mapped to priority 3 and 4, respectively.
+
+### Scope
+
+The purpose of this test plan is to test the PFC PAUSE processing behavior of a SONiC DUT and its capability to pause or un-pause traffic with right priorities. The test assumes all necessary configuration is already pre-configured on the SONIC DUT before test runs.
+
+### Testbed
+
+```
++-------------+      +--------------+      +-------------+
+| Keysight TX |------|   SONiC DUT  |------| Keysight RX |
++-------------+      +--------------+      +-------------+
+
+Keysight ports are connected with SONiC switch as shown in the illustration above.
+```
+## Setup configuration
+
+### DUT configuration
+
+- PFC watch dog is disabled
```

Comment:
watch dog -> watchdog. Add a reference to PFC watchdog, e.g., https://github.com/Azure/SONiC/wiki/PFC-Watchdog-Design. If necessary, please elaborate why PFC watchdog should be disabled.